connmgr/connmanager.go

```diff
@@ -307,12 +302,12 @@ out:
 			case registerPending:
 				connReq := msg.c
 				connReq.updateState(ConnPending)
-				pending[msg.c.id] = connReq
+				pending[msg.c.id.Load()] = connReq
```

Comment:
Previously the loop inside the `connHandler` function was accessing the `ConnReq.id` primitive directly rather than via atomic funcs.